runtime/src/bank.rs

```diff
@@ -2509,14 +2513,14 @@ impl Bank {
             dbank.is_delta.load(Ordering::Relaxed)
         );
 
-        let st = self.stakes.read().unwrap();
-        let dst = dbank.stakes.read().unwrap();
-        assert_eq!(*st, *dst);
-
         let bh = self.hash.read().unwrap();
         let dbh = dbank.hash.read().unwrap();
         assert_eq!(*bh, *dbh);
 
+        let st = self.stakes.read().unwrap();
```

Comment:
@BurtonQin nits: reordering is alone suffice to fix the deadlock as you said. Thanks for spotting this!

But for maintainability perspective, the current code is too fragile, relaying on the correct locking order. I think this method's each code blocks should also be surrounded with `{}` so that each locks aren't held for longer than needed.